wallet/src/Pos/Wallet/Web/Methods/History.hs

```diff
@@ -202,17 +203,19 @@ updateTransaction :: MonadWalletWebMode m => AccountId -> CTxId -> CTxMeta -> m
 updateTransaction accId txId txMeta = do
     setWalletTxMeta (aiWId accId) txId txMeta
 
-addRecentPtxHistory
+addPtxHistory
     :: MonadWalletWebMode m
     => CId Wal -> Map TxId TxHistoryEntry -> m (Map TxId TxHistoryEntry)
-addRecentPtxHistory wid currentHistory = do
+addPtxHistory wid currentHistory = do
     pendingTxs <- fromMaybe [] <$> getWalletPendingTxs wid
     let conditions = map _ptxCond pendingTxs
     -- show only actually pending transactions in logs
     logTxHistory "Pending" $ mapMaybe (preview _PtxApplying) conditions
     -- but return all transactions which are not yet in blocks
-    let candidatesList = txHistoryListToMap (mapMaybe ptxPoolInfo conditions)
+    let candidatesList = txHistoryListToMap (mapMaybe getPtxTh conditions)
     return $ Map.union currentHistory candidatesList
+  where
+    getPtxTh cond = guard (isPtxActive cond) *> ptxPoolInfo cond
```

Comment:
Right, this is effectively filtering the history to only Txs which are anything but in `Creating` state.